src/ContextMenu.tsx

```diff
@@ -23,24 +23,38 @@ import { Box, Break, IBox, Text, useTheme } from './';
  */
 
 interface IContextMenuProps {
-  renderTrigger?: (props?: IContextMenuProps) => ReactNode | string;
+  renderTrigger: (
+    ref: React.MutableRefObject<{
+      handleContextClick: (e: MouseEvent<HTMLDivElement>) => void;
+      // TODO get event type
+      handleContextMenu: (e: any) => void;
+      handleMouseDown: (e: MouseEvent<HTMLDivElement>) => void;
+      handleMouseOut: (e: MouseEvent<HTMLDivElement>) => void;
+      handleMouseUp: (e: MouseEvent<HTMLDivElement>) => void;
+      handleTouchEnd: (e: TouchEvent<HTMLDivElement>) => void;
+      handleTouchstart: (e: TouchEvent<HTMLDivElement>) => void;
+    }>
+  ) => ReactNode | string;
 }
 
 export interface IContextMenu extends IContextMenuProps, IContextMenuViewProps {}
 
 export const ContextMenu: FunctionComponent<IContextMenu> = props => {
   const { id, renderTrigger, ...rest } = props;
 
+  const contextTriggerRef = React.useRef<any>(null);
+
   return (
     <Fragment>
-      {renderTrigger && <ReactContextMenuTrigger id={id}>{renderTrigger()}</ReactContextMenuTrigger>}
-
+      <ContextMenuTrigger id={id} ref={contextTriggerRef} holdToDisplay={-1}>
+        {renderTrigger(contextTriggerRef)}
```

Comment:
The `contextTriggerRef.current` passed to renderTrigger going to be null initially unless `ContextMenuTrigger` is rendered (which happens after renderTrigger is executed)
Is this intended? I don't think so.

Reply:
renderTrigger is now required, `ContextMenuTrigger` should always be rendered

Reply:
@P0lip I saw in studio there was an area where we used the trigger and menu separately, but that was done as a workaround for missing functionality (open on clicking the trigger) here. I can't think of a reasonable scenario where we would want renderTrigger to optional

Reply:
I am afraid we may have no reasonable solution for the given case.
We can either not pass ref at all (if one need ref, ContextMenuTrigger can be embedded manually) or simply handle that null in renderTrigger.
Both are some intermediate solution to the problem, so maybe let's leave it as it?
Just make sure the typings are correct.

Reply:
I say we leave it. Rendering ContextMenuTrigger separately is kind of messy. Will update the ref type to be possibly undefined so that its clearer to the end user